### DISCUSSION TITLE
Skip beforeEach

### DISCUSSION THREAD

**Discussion by craigpotter** [pestphp/pest#397] [Q&A] [answered]
Hi, 

Is it possible to skip a beforeEach() for a certain test in a file? 

Say I have a CustomerTest.php for an admin panel. 

In the beforeEach() I might have some seeding or set up but then if I want to quickly test that it can't not be accessed by unauthorized user or some other simple non data related test, there is no need for the setup/seeding or maybe minimal setup required. 

I wouldn't want to create a Test file for this small test but also the setup slows the tests down when you have a few in each test file. 

I might have 20 tests in a file and 3/4 could easily be skip the beforeEach method and still be good tests.  

Might be a better way to do it with groups maybe or I could just split them out I guess. 

Just wonder if you smart folk have any ideas.

**Comment by pildit** [comment, marked as answer]
You can use `$this->getName()` in beforeEach to get the name of the current method
